### PR TITLE
[Bug Fix] Correct PostSummonPhase Timing for Abilities and Entry Hazards

### PR DESCRIPTION
**Pull request by Greenlamp2**
## What are the changes?

This PR introduces a fix to the handling of the PostSummonPhase for abilities like Intimidate and entry hazards. It ensures that the phase is triggered correctly after a Pokémon is switched into battle.

### Before:
- The PostSummonPhase for the opponent was triggered prematurely, causing issues with ability and entry hazard activation.
- A solution was implemented, but did not work for subsequent switches.

### After:
- Introduced a conditional phase execution mechanism.
- The PostSummonPhase is now queued and only executed when specific conditions (like the correct number of Pokémon on the field) are met.
- This ensures abilities like Intimidate and entry hazards trigger correctly after switches.

### Screenshots/Videos
webstorm64_qSUF6IKmmf
webstorm64_fms8eDtfAT

## How to test the changes?
Run all unit tests (`npm run test`) to ensure no existing functionality is broken.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes? (N/A)

### CODE REVIEW

**Comment by flx-sta** [comment]
you could group single & double tests with another `describe`

**Comment by Greenlamp2** [comment]
> you could group single & double tests with another `describe`

yep, but it will be for another time, i don't have any time left today, i wanted to fix the entry hasard issue before leaving

**Review by flx-sta — APPROVED** [review]
lgtm 🚀

**Review by Tempo-anon — APPROVED** [review]

File: src/phases.ts [hunk 1]
```diff
@@ -1024,6 +1024,15 @@ export class EncounterPhase extends BattlePhase {
     });
 
     if (this.scene.currentBattle.battleType !== BattleType.TRAINER) {
+      enemyField.map(p => this.scene.pushConditionalPhase(new PostSummonPhase(this.scene, p.getBattlerIndex()), () => {
+        // is the player party initialized ?
+        const a = !!this.scene.getParty()?.length;
+        // how many player pokemon are on the field ?
+        const amountOnTheField = this.scene.getParty().filter(p => p.isOnField()).length;
+        // if it's a double, there should be 2, otherwise 1
```
Comment (Tempo-anon) [inline src/phases.ts:1032]: nit: not sure if comments are really needed if the code is understandable

**Pull request merged**